src/AppScopeTrait.php

```diff
@@ -3,5 +3,43 @@
 namespace atk4\core;
 
 trait AppScopeTrait {
+
+    public $_appScope = true;
+
+    /**
+     * Always points to current Application
+     *
+     * @var App_CLI
+     */
     public $app;
+
+    /**
+     * Object in Agile Toolkit contain $name property which is derrived from
+     * the owher object and keeps extending as you add objects deeper into
+     * run-time tree. Sometimes that may generate long names. Long names are
+     * difficult to read, they increase HTML output size but most importantly
+     * they may be restricted by security extensions such as SUHOSIN.
+     *
+     * Agile Toolkit implements a mechanism which will replace common beginning
+     * of objects with an abbreviation thus keeping object name length under
+     * control. This variable defines the maximum length of the object's $name.
+     * Be mindful that some objects will concatinate theri name with fields,
+     * so the maximum letgth of GET argument names can exceed this value by
+     * the length of your field.
+     *
+     * We recommend you to increase SUHOSIN get limits if you encounter any
+     * problems. Set this value to "false" to turn off name shortening.
```

Comment:
Link to Suhosin manual would be nice here. Or at least to StackOverflow like this: http://stackoverflow.com/a/9399615/1466341